src/dsp_tools/commands/start_stack.py

```diff
@@ -323,28 +333,39 @@ def _start_remaining_docker_containers(self) -> None:
         """
         compose_str = "docker compose -f docker-compose.yml"
         if self.__stack_configuration.latest_dev_version:
+            logger.debug("Running 'docker compose pull' ...")
             subprocess.run("docker compose pull".split(), cwd=self.__docker_path_of_user, check=True)
             compose_str += " -f docker-compose.override.yml"
         if self.__stack_configuration.custom_host is not None:
             compose_str += " -f docker-compose.override-host.yml"
         compose_str += " up -d"
+        logger.debug(f"Running '{compose_str}' ...")
         subprocess.run(compose_str.split(), cwd=self.__docker_path_of_user, check=True)
 
     def _wait_for_api(self) -> None:
         """
         Wait until the API is up and running.
         This mimicks the behaviour of the script webapi/scripts/wait-for-api.sh in the DSP-API repository.
         """
+        logger.debug("Waiting for the API to start...")
         for _ in range(6 * 60):
             try:
-                response = requests.get(f"{self.__localhost_url}:3333/health", timeout=1)
-                if not response.ok:
+                params = RequestParameters("GET", f"{self.__localhost_url}:3333/health", timeout=1)
+                log_request(params)
+                response = requests.get(params.url, timeout=params.timeout)
+                log_response(response)
+                if response.ok:
+                    break
```

Comment:
I'm very puzzled about this. This fixes an obvious bug: As soon as the container is healthy, the loop must break. But why did it work before on the main branch? The legacy-e2e tests never took 6 mins (as I would have expected), but rather 2-3 mins.

This means that in fact the loop has never taken 6 mins. But I don't see a possibility in the code how this could happen!

Reply:
For future reference: Before, the healthcheck route was called 6 * 60 times. As long as the API was not healthy yet, it slept 1 s, but as soon as the API became healthy, it didn't sleep any more. This means that the remaining healthcheck calls were very fast.